misc/package/Data Files/MWSE/core/lib/mwse_safeObjectHandle.lua

```diff
@@ -15,72 +18,89 @@ end
 event.register("objectInvalidated", onObjectInvalidated)
 
 -- Create a new unsafe reference.
-function this.new(object)
+function SafeHandle.new(object)
 	if (object == nil) then
-		return nil
+		return
 	end
 
-	-- Return a previous handle if applicable.
-	if (handles[object]) then
-		return handles[object]
+	-- Create a new handle if necessary.
+	if handles[object] == nil then 
+		handles[object] = setmetatable({ _object = object }, handleMetatable)
 	end
 
-	local raw = { _object = object }
-	function raw:valid()
-		local object = rawget(self, "_object")
-		return object ~= nil and not object.deleted
-	end
-	function raw:getObject() return rawget(self, "_object") end
+	-- Return the handle for this object.
+	return handles[object]
+end
+
+function SafeHandle:isValid()
+	local object = rawget(self, "_object")
+	return object ~= nil and not object.deleted
+end
 
-	-- Create a metatable redirect.
-	local handle = setmetatable(raw, this)
-	handles[object] = handle
+-- Backwards compatibility
+SafeHandle.valid = SafeHandle.isValid
 
-	return handle
+function SafeHandle:getObject()
+	return rawget(self, "_object")
 end
 
--- Allow this handle to index into the reference, with some checking.
-function this:__index(key)
-	local valueRaw = rawget(self, key)
-	if (valueRaw) then
-		return valueRaw
-	end
+-- Fill out the metatable.
+
+function handleMetatable.__index(handle, key)
+	-- Try to look things up in the `SafeHandle` class.
+	local val = SafeHandle[key]
```

Comment:
The `SafeHandle` table does not have a metatable, so there's no need to use `rawget` here. (Only objects created by `SafeHandle.new` have metatables.)